fastlane_core/lib/fastlane_core/cert_checker.rb

```diff
@@ -144,29 +144,71 @@ def self.install_missing_wwdr_certificates
       # Install all Worldwide Developer Relations Intermediate Certificates listed here: https://www.apple.com/certificateauthority/
       missing = WWDRCA_CERTIFICATES.map { |c| c[:alias] } - installed_wwdr_certificates
       missing.each do |cert_alias|
-        install_wwdr_certificate(cert_alias)
+        url = WWDRCA_CERTIFICATES.find { |c| c[:alias] == cert_alias }.fetch(:url)
+        file = Tempfile.new(File.basename(url))
+
+        unless fetch_certificate(url, file)
+          UI.verbose("Could not fetch certificate #{cert_alias}")
+          next
+        end
+
+        unless check_expiry(filename)
+         puts "#{filename} is invalid"
+         UI.verbose("Skipping installation of certificate: #{filename}") 
+         next
+        end
+
+        puts "Installing #{cert_alias}"
+        import_wwdr_certificate(filename)
       end
       missing.count
     end
 
-    def self.install_wwdr_certificate(cert_alias)
-      url = WWDRCA_CERTIFICATES.find { |c| c[:alias] == cert_alias }.fetch(:url)
-      file = Tempfile.new(File.basename(url))
-      filename = file.path
-      keychain = wwdr_keychain
-      keychain = "-k #{keychain.shellescape}" unless keychain.empty?
+    def self.fetch_certificate(url, filename)
+        # Attempts to fix an issue installing WWDR cert tends to fail on CIs
+        # https://github.com/fastlane/fastlane/issues/20960
+        curl_extras = ""
+        if FastlaneCore::Feature.enabled?('FASTLANE_WWDR_USE_HTTP1_AND_RETRIES')
+          curl_extras = "--http1.1 --retry 3 --retry-all-errors "
+        end
+
+        fetch_command = "curl #{curl_extras}-f -o #{filename} #{url}"
+        require 'open3'
+        stdout, stderr, status = Open3.capture3(fetch_command)
+        if FastlaneCore::Globals.verbose?
+          UI.command_output(stdout)
+          UI.command_output(stderr)
+        end
+        unless status.success?
+          return false
+        end
+        return true
+    end
 
-      # Attempts to fix an issue installing WWDR cert tends to fail on CIs
-      # https://github.com/fastlane/fastlane/issues/20960
-      curl_extras = ""
-      if FastlaneCore::Feature.enabled?('FASTLANE_WWDR_USE_HTTP1_AND_RETRIES')
-        curl_extras = "--http1.1 --retry 3 --retry-all-errors "
+    def self.check_expiry(filename)
+
+      check_command = "security verify-cert -c #{filename}"
+      UI.verbose("Checking expiry date of certificate")
+      require 'open3'
+      stdout, stderr, = Open3.capture3(check_command)
```

Comment:
Like above, conventions and security practices recommend to use `Open3.capture3('security', 'verify-cert', '-c', filename)` with the command arguments already split and passed individually rather than passing a single string and letting the shell interpreter split it into arguments based on spaces and quotes. 🙃 

(And same for `security import` on lie 209 below ofc)